datafusion/core/tests/sqllogictests/test_files/interval.slt

```diff
@@ -182,3 +182,193 @@ from t;
+query D
+select '1980-01-01'::date + interval '1 day'
+----
+1980-01-02
+
+
+query P
+select '1980-01-01'::timestamp + interval '1 day'
+----
+1980-01-02T00:00:00
+
+
+### date / timestamp (scalar) - interval (scalar)
+query D
+select '1980-01-01'::date - interval '1 day'
+----
+1979-12-31
+
+
+query P
+select '1980-01-01'::timestamp - interval '1 day'
+----
+1979-12-31T00:00:00
+
+
+### date / timestamp (array) + interval (scalar)
+query D
+select d + interval '1 day' from t;
+----
+1980-01-02
+1990-10-02
+1980-01-03
+
+query P
+select ts + interval '1 day' from t;
+----
+2000-01-02T00:00:00
+2000-01-02T12:11:10
+2000-02-02T00:00:00
+
+### date / timestamp (array) - interval (scalar)
+query D
+select d - interval '1 day' from t;
+----
+1979-12-31
+1990-09-30
+1980-01-01
+
+query P
+select ts - interval '1 day' from t;
+----
+1999-12-31T00:00:00
+1999-12-31T12:11:10
+2000-01-31T00:00:00
+
+### date / timestamp (scalar) + interval (array)
+query error DataFusion error: Internal error: If RHS of the operation is an array, then LHS also must be\. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
+select '1980-01-01'::date + i from t;
+
+query error DataFusion error: Internal error: If RHS of the operation is an array, then LHS also must be\. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
+select '1980-01-01T12:00:00'::timestamp + i from t;
+
+
+query error DataFusion error: Internal error: If RHS of the operation is an array, then LHS also must be\. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
+select '1980-01-01'::date - i from t;
+
+query error DataFusion error: Internal error: If RHS of the operation is an array, then LHS also must be\. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
+select '1980-01-01T12:00:00'::timestamp - i from t;
+
+### date / timestamp (array) + interval (array)
+query D
+select d + i from t;
+----
+1980-02-01
+1990-10-02
+1980-01-02
+
+query P
+select ts + i from t;
+----
+2000-02-01T00:00:00
+2000-01-02T12:11:10
+2000-02-01T00:01:00
+
+
+### date / timestamp (array) - interval (array)
+query D
+select d - i from t;
+----
+1979-12-01
+1990-09-30
+1980-01-02
+
+query P
+select ts - i from t;
+----
+1999-12-01T00:00:00
+1999-12-31T12:11:10
+2000-01-31T23:59:00
+
+
+# Now reverse the argument order
+# interval (scalar) + date / timestamp (scalar)
+query D
+select '1 month'::interval + '1980-01-01'::date;
+----
+1980-02-01
+
+query P
+select '1 month'::interval + '1980-01-01T12:00:00'::timestamp;
+----
+1980-02-01T12:00:00
+
+# Exected error: interval (scalar) - date / timestamp (scalar)
+
+query error DataFusion error: Error during planning: interval can't subtract timestamp/date
```

Comment:
This is more a wish list item: I think the error message is only half helpful:

```suggestion
query error DataFusion error: Error during planning: interval can't subtract date from interval
```

Reply:
I agree -- the error messages should be better. I will update them in a follow on PR

Reply:
Improved messages in https://github.com/apache/arrow-datafusion/pull/6230